File: backend/app/schemas/cash_book.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class CashBookItem(BaseModel):
    """
    Individual Cash Ledger Transaction Item for Cash Book table.
    """
    model_config = ConfigDict(from_attributes=True)

    id: str
    date: datetime
    formatted_date: str
    formatted_time: str
    description: str
    code: str = "—"
    name: str = "—"
    invoice: str = "—"
    transaction_type: str = Field(
        ...,
        description="Type: opening_balance, cash_sale, collection, expense",
    )
    debit: float = Field(0.0, description="Cash OUT")
    credit: float = Field(0.0, description="Cash IN")
    balance: float = Field(..., description="Running Cash in Hand after transaction")


class CashBookSummary(BaseModel):
    """
    Complete Cash Book Response Payload for a selected business date or range.
    """
    model_config = ConfigDict(from_attributes=True)

    date: str
    start_date: datetime
    end_date: datetime
    timezone: str
    currency_symbol: str
    
    # Financial KPI summary cards
    previous_balance: float = Field(..., description="Opening cash before this date")
    today_cash_received: float = Field(..., description="Cash inflows during period")
    today_cash_expense: float = Field(..., description="Actual cash expenses during period (Expense module only)")
    total_expense: float = Field(0.0, description="Actual cash expenses during period (Expense module only)")
    total_purchase_paid: float = Field(0.0, description="Cash paid on purchases during period")
    total_supplier_paid: float = Field(0.0, description="Cash paid to suppliers during period")
    total_refund_paid: float = Field(0.0, description="Cash refunded on returns during period")
    cash_in_hand: float = Field(..., description="Closing cash in hand = previous + received - total_paid")
    
    total_cash_received: float = Field(..., description="Total cash received in period")
    total_cash_paid: float = Field(..., description="Total cash paid in period (all outflows)")
    closing_cash_balance: float = Field(..., description="Closing cash balance")

    # Business profile info for print / branding
    company_name: Optional[str] = None
    company_address: Optional[str] = None
    company_phone: Optional[str] = None
    company_logo: Optional[str] = None

    items: List[CashBookItem] = Field(default_factory=list)

File: frontend/src/types/cash_book.ts
export interface CashBookItem {
  id: string;
  date: string;
  formatted_date: string;
  formatted_time: string;
  description: string;
  code: string;
  name: string;
  invoice: string;
  transaction_type:
    | "opening_balance"
    | "cash_sale"
    | "collection"
    | "expense";
  debit: number;
  credit: number;
  balance: number;
}

export interface CashBookSummary {
  date: string;
  start_date: string;
  end_date: string;
  timezone: string;
  currency_symbol: string;
  previous_balance: number;
  today_cash_received: number;
  today_cash_expense: number;
  total_expense?: number;
  total_purchase_paid?: number;
  total_supplier_paid?: number;
  total_refund_paid?: number;
  cash_in_hand: number;
  total_cash_received: number;
  total_cash_paid: number;
  closing_cash_balance: number;
  company_name?: string | null;
  company_address?: string | null;
  company_phone?: string | null;
  company_logo?: string | null;
  items: CashBookItem[];
}
